Source/DafnyLanguageServer.Test/Util/ClientBasedLanguageServerTest.cs

```diff
@@ -20,16 +20,13 @@ public class ClientBasedLanguageServerTest : DafnyLanguageServerTestBase {
   protected ILanguageClient client;
   protected DiagnosticsReceiver diagnosticsReceiver;
 
-  public async Task<Diagnostic[]> GetLastVerificationDiagnostics(TextDocumentItem documentItem, CancellationToken cancellationToken = default, int? expectedNumber = null) {
+  public async Task<Diagnostic[]> GetLastDiagnostics(TextDocumentItem documentItem, CancellationToken cancellationToken = default) {
     await client.WaitForNotificationCompletionAsync(documentItem.Uri, cancellationToken);
-    var document = await Documents.GetVerifiedDocumentAsync(documentItem);
-    await diagnosticsReceiver.AwaitNextDiagnosticsAsync(cancellationToken); // Get resolution diagnostics
-    var remainingDiagnostics = expectedNumber ?? Int32.MaxValue;
+    var document = await Documents.GetLastDocumentAsync(documentItem);
     Diagnostic[] result;
     do {
       result = await diagnosticsReceiver.AwaitNextDiagnosticsAsync(cancellationToken);
-      remainingDiagnostics--;
-    } while (!document!.Diagnostics.SequenceEqual(result) && remainingDiagnostics > 0);
+    } while (!document!.Diagnostics.SequenceEqual(result));
```

Comment:
Why do you need this? It's a bit smelly to access the diagnostics of the document and compare them against what was received, because it could or could not have been updated in the meanwhile.

Reply:
We're comparing the diagnostics of the final document, which won't be updated anymore, against the diagnostics of the document.

Reply:
Very good insight, so as long as we call this method after every change, and call it until there are no more diagnostics, we should be fine.